packages/data-api/src/AnalyticsFetchQuery.js

```diff
@@ -144,58 +155,43 @@ export class AnalyticsFetchQuery extends DataFetchQuery {
+    const relationsTableName = `entity_relations_a${aggregation.stackId}`;
+    return `INNER JOIN ${relationsTableName} on ${relationsTableName}.code = ${previousTableName}.entity_code`;
+  }
 
-    const groupByFields = [...this.getCommonFields()];
-    const { groupByPeriodField } = this.aggregation.switches;
+  getAggregationGroupByClause(aggregation) {
+    const groupByFields = [this.getEntityCodeField(aggregation), 'data_element_code'];
+    const { groupByPeriodField } = aggregation.switches;
     if (groupByPeriodField) groupByFields.push(groupByPeriodField);
     return `GROUP BY ${groupByFields.join(', ')}`;
-  };
-
-  getLatestPerPeriodClause() {
-    if (!this.isAggregating || !this.aggregation.switches.getLatestPerPeriod) return '';
-
-    // add where condition for each non-calculated table selected in a1
-    const joinFields = [...this.getCommonFields()];
-    const { groupByPeriodField } = this.aggregation.switches;
-    if (groupByPeriodField) joinFields.push(groupByPeriodField);
-    const joinConditions = joinFields.map(field => `${field} = a1.${field}`);
-
-    // add start/end date clauses to limit results and speed things up
-    const { periodConditions, periodParams } = this.getPeriodConditionsAndParams();
-    this.paramsArray.push(...periodParams);
+  }
 
-    const conditions = [...joinConditions, ...periodConditions];
+  build() {
+    const commonTableExpressions = `${this.aggregations
+      .map(aggregation => this.getEntityCommonTableExpression(aggregation))
+      .join('\n')}`;
 
-    const fields = this.aggregation.switches.sum
-      ? ANSWER_SPECIFIC_FIELDS.filter(field => field !== 'value')
-      : ANSWER_SPECIFIC_FIELDS;
+    const baseAnalytics = `(SELECT *, day_period as period from analytics
+      ${this.getBaseInnerJoins()}
+      ${this.getBaseWhereClause()}) as base_analytics`;
 
-    return `
-      CROSS JOIN LATERAL (
-        SELECT ${fields}
-        FROM analytics
-        INNER JOIN entity_codes_and_relations ON entity_codes_and_relations.code = analytics.entity_code
-        WHERE ${conditions.join('\n      AND ')}
-        ORDER BY date DESC
-        LIMIT 1
-      ) as latestPerPeriod
-    `;
-  }
+    const wrapAnalyticsInAggregation = (analytics, aggregation) =>
+      `(${this.getAggregationSelect(aggregation)} 
+      FROM 
+      ${analytics}
+      ${this.getAggregationJoin(aggregation)}
+      ${this.getAggregationGroupByClause(aggregation)}) as a${aggregation.stackId}`;
 
-  build() {
     this.query = `
-      ${this.getEntityCommonTableExpression()}
+      ${commonTableExpressions}
       SELECT ${this.getAliasedColumns()}
-      FROM (
-        ${this.getA1Select()}
-        FROM analytics
-          ${this.getA1InnerJoins()}
-          ${this.getA1WhereClause()}
-          ${this.getA1GroupByClause()}
-      ) as a1
-      ${this.getLatestPerPeriodClause()}
+      FROM ${this.aggregations.reduce(wrapAnalyticsInAggregation, baseAnalytics)}
```

Comment:
While this query builder reads to me a lot more simply than the previous iteration, it still has it's tricks about it 😢 Food for a walk-n-talk I think...